File: java/src/main/java/devtools/handlebars/HandlebarsViewResolver.java
package devtools.handlebars;

import com.github.jknack.handlebars.Handlebars;
import org.springframework.web.servlet.view.AbstractTemplateViewResolver;
import org.springframework.web.servlet.view.AbstractUrlBasedView;

public class HandlebarsViewResolver extends AbstractTemplateViewResolver {

    final Handlebars handlebars;

    public HandlebarsViewResolver(Handlebars handlebars) {
        this.handlebars = handlebars;
        setContentType("text/html;charset=UTF-8");
        setViewClass(requiredViewClass());
    }


    @Override
    protected AbstractUrlBasedView buildView(String viewName) throws Exception {

        if( viewName == null || viewName.trim().isEmpty()) {
            viewName = "index";
        }

        HandlebarsView view = (HandlebarsView) super.buildView(viewName);
        view.template = handlebars.compile(viewName);
        return view;
    }


    @Override
    protected Class<?> requiredViewClass() {
        return HandlebarsView.class;
    }

    public HandlebarsViewResolver registerHelpers(final Object helper) {
        handlebars.registerHelpers(helper);
        return this;
    }

}

File: src/main/kotlin/devtools/handlebars/HandlebarsViewResolver.kt
package devtools.handlebars

import com.github.jknack.handlebars.Handlebars
import org.springframework.web.servlet.view.AbstractTemplateViewResolver
import org.springframework.web.servlet.view.AbstractUrlBasedView

class HandlebarsViewResolver: AbstractTemplateViewResolver {

    lateinit var handlebars: Handlebars

    constructor() {
        viewClass = requiredViewClass()
        contentType = "text/html;charset=UTF-8"
    }


    override fun buildView(viewName: String): AbstractUrlBasedView {
        val view = super.buildView(viewName) as HandlebarsView
        view.template = handlebars.compile(viewName)
        return view
    }

    override fun requiredViewClass(): Class<*> {
        return HandlebarsView::class.java
    }

    fun registerHelpers(bean: Any): HandlebarsViewResolver {
        handlebars.registerHelpers(bean)
        return this
    }
}

